### PR TITLE
Add ToggleAdvanced widget documentation

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
## Purpose of this pull request

This pull request (PR) adds information about the ToggleAdvanced widget.

## Affected DevDocs pages

These pages do not exist yet, but they will be created for the ToggleAdvanced widget documentation and the URLs should be like the following:

- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_toggle.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget_toggle.html
- https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/widgets/widget_toggle.html

## Links to Magento source code

- https://github.com/magento/magento2/blob/2.1/lib/web/mage/toggle.js - Toggle Widget source code
- https://github.com/magento/magento2/blob/2.2/lib/web/mage/toggle.js - Toggle Widget source code
- https://github.com/magento/magento2/blob/2.3/lib/web/mage/toggle.js - Toggle Widget source code
- https://github.com/magento/magento2/blob/2.1/app/code/Magento/CatalogInventory/view/frontend/templates/stockqty/composite.phtml#L16 - Example of usage
- https://github.com/magento/magento2/blob/2.2/app/code/Magento/CatalogInventory/view/frontend/templates/stockqty/composite.phtml#L16 - Example of usage
- https://github.com/magento/magento2/blob/2.3/app/code/Magento/CatalogInventory/view/frontend/templates/stockqty/composite.phtml#L16 - Example of usage

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+For information about how to initialize a widget in a JS component or `.phtml` template, see the [Initialize JavaScript] topic.
+
+Generally the ToggleAdvanced widget is instantiated like following:
+
+```javascript
+$("#element").toggleAdvanced();
+```
+
+Where:
+-   `#element` is the selector of the element for ToggleAdvanced is initialized.
```
Comment (rogyar) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]: Please, adjust slightly 
```suggestion
-   `#element` is the selector of the element for which ToggleAdvanced is initialized.
```

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+The class used to be toggled on clicked element.
+
+**Type**: String
+
+**Default value**: `active`
+
+### `selectorsToggleClass`
+
+The class used to be toggled on selectors DOM elements.
```
Comment (rogyar) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]: Please, adjust in the following way
```suggestion
The class that will be toggled (added/removed) for selected DOM elements.
```

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+**Type**: String
+
+**Default value**: `hidden`
+
+### `toggleContainers`
+
+The selectors for DOM elements we wish to toggle.
```
Comment (rogyar) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]: Please, adjust slightly
```suggestion
The selectors of DOM elements that will be toggled.
```

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+Text of the old label to be used on toggle.
+
+**Type**: String
+
+**Default value**: `null`
+
+### `currentLabelElement`
+
+Current label of the container.
```
Comment (rogyar) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]: I believe that it's rather a container of the current label element than the label of the container.
```suggestion
Container element of the current label
```

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    </div>
+    <div class="toggle-container" id="toggleContainer">
+        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad autem, corporis ducimus ex in ipsam minima nisi nulla officiis omnis quae quaerat quasi quo ratione reiciendis rem repellendus temporibus, voluptates.</p>
+    </div>
+</div>
+```
+
+## Result
+
+The result is two sections with initial content and not toggled content.
```
Comment (rogyar) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]: not toggled = toggled? :)

**Comment by rogyar** [comment]
Hi @serhiyzhovnir. Thank you for the great topic. Please, provide the minor adjustments mentioned in the review. 
Thank you

**Comment by serhiyzhovnir** [comment]
Hi @rogyar 
We have added the required adjustments for this PR.
Can you, please, check it?
Thank you!

**Comment by dobooth** [comment]
@serhiyzhovnir Thanks for the contribution. I need to ask you to make a change or two.
First: No need to add the same image multiple time. They should be moved to `common/images` and then linked from there.
Second: Run them through an image optimizer such as https://imagecompressor.com/ to get the file size down. 
The best: Since you have committed these images already, they will remain in the Git history, needlessly increasing the size of the repo. If you could make a new PR with the optimized images in the correct place, that would be amazing.

**Comment by serhiyzhovnir** [comment]
Hi @dobooth
We have created new PR with requested changes: https://github.com/magento/devdocs/pull/4780
Thank you!

**Pull request closed without merging**